platforms/core-configuration/model-reflect/src/test/groovy/org/gradle/internal/reflect/annotations/impl/DefaultTypeAnnotationMetadataStoreTest.groovy

```diff
@@ -1009,6 +1068,13 @@ class DefaultTypeAnnotationMetadataStoreTest extends Specification implements Va
         def expectedPropertyNames = expectedProperties.keySet().sort()
         assert actualPropertyNames == expectedPropertyNames
 
+        def validationContext = DefaultTypeValidationContext.withoutRootType(false, TestUtil.problemsService())
+        metadata.visitValidationFailures(validationContext)
+        List<String> actualErrors = validationContext.problems
+            .collect({ (normaliseLineSeparators(TypeValidationProblemRenderer.renderMinimalInformationAbout(it)) + (it.definition.severity == Severity.ERROR ? " [STRICT]" : "") as String) })
+        actualErrors.sort()
+        println actualErrors.join("\n--- 8< -------------------------\n")
```

Comment:
💭 Moved this up here so we print the actual errors received _before_ we potentially fail the checks below, so it's always easy to see what the reported problems are.